src/services/Transaction/transaction.js

```diff
@@ -17,9 +17,28 @@ export async function postTransaction(accountId, transactionData) {
     }
 }
 
-export function getTransactionsQuery(accountId) {
-    return query(
+export function getTransactionsQuery(accountId, filter = null) {
+    let queryConstrtaints = [];
```

Comment:
Same here: you can declare an array as a const and still push new items, the push function just exapnd the memory allocated for that array doest not change the actual memory address